builtin/credential/oidc/path_config.go

```diff
@@ -0,0 +1,275 @@
+	if conf.IssuerCABundle != "" {
+		bundles, err := certPoolFromString(conf.IssuerCABundle)
+		if err != nil {
+			return nil, err
+		}
+		transport := &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: bundles,
+			},
+		}
+		httpClient := &http.Client{
+			Transport: transport,
+		}
+		ctx = oidc.ClientContext(ctx, httpClient)
+	}
+
+	return oidc.NewProvider(ctx, conf.IssuerUrl)
+}
+
+func (b *backend) validateAndExtractClaims(cfg *ConfigEntry, rawToken string) (userName string, groups []string, token *oidc.IDToken, err error) {
+	provider, err := b.oidcProviderForConfig(cfg)
+	if err != nil {
+		return "", []string{}, nil, err
+	}
+	if provider == nil {
+		return "", []string{}, nil, errors.New("OIDC provider not configured")
+	}
+	verifier := provider.Verifier(&oidc.Config{SkipClientIDCheck: true}) // note we verify the audience ourselves.
+	idToken, err := verifier.Verify(context.TODO(), rawToken)
+	if err != nil {
+		return "", []string{}, nil, errors.New("OIDC identity token error: " + err.Error())
+	}
+	if !containsAny(idToken.Audience, cfg.ClientIDs) {
+		b.Logger().Info("OIDC token has bad 'aud': %v", idToken.Audience)
+		return "", []string{}, nil, errors.New("OIDC identity token issued for unsupported client ID.")
+	}
+	claims := make(map[string]interface{})
+	if err := idToken.Claims(&claims); err != nil {
+		b.Logger().Warn("OIDC token claims parsing error: %v", err)
+		return "", []string{}, nil, errors.New("OIDC identity token claim parsing error.")
+	}
+	b.Logger().Debug("OIDC claims %v", claims) // TODO(mwitkow): Remove.
+
+	userNameClaim, ok := claims[cfg.UsernameClaim].(string)
+	if !ok {
+		b.Logger().Warn("OIDC token doesn't have username under expected claim. Claims: %v", claims)
+		return "", []string{}, nil, errors.New("OIDC identity token user claim parsing error.")
+	}
+	userName = userNameClaim
+	if cfg.GroupsClaim != "" {
+		groupNameClaim, ok := claims[cfg.GroupsClaim].([]interface{})
+		if !ok {
+			b.Logger().Warn("OIDC token doesn't have group name under expected claim.", "claimName", cfg.GroupsClaim, "allClaims", claims)
+			return "", []string{}, nil, errors.New("OIDC identity token group claim parsing error.")
+		}
+		for _, gInt := range groupNameClaim {
+			g, ok := gInt.(string)
+			if ok {
+				groups = append(groups, g)
+			}
+		}
+	}
+	return userName, groups, idToken, nil
+}
+
+func containsAny(existingValues []string, checkedValues []string) bool {
+	for _, s := range existingValues {
+		for _, c := range checkedValues {
+			if s == c {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func certPoolFromString(caBundle string) (*x509.CertPool, error) {
+	pool := x509.NewCertPool()
+	ok := pool.AppendCertsFromPEM([]byte(caBundle))
+	if !ok {
+		return nil, errors.New("failed parsing CA bundle")
+	}
+	return pool, nil
+}
+
+// ConfigEntry for OIDC configuration.
+type ConfigEntry struct {
+	IssuerUrl      string   `json:"issuer_url"`
+	IssuerCABundle string   `json:"issuer_verify_ca"`
```

Comment:
If this is inlined instead of referring to a path you can probably just use a []byte. Are there any other examples of vault doing this?